### PR TITLE
Abdel deadline follow up tracking solution

### PR DESCRIPTION
**Pull request by abdel-lall**
# Description
1.	#### $\textcolor{red}{\textsf{(PRIORITY HIGH) Jae:}}$  Create way for me to track people I’ve followed up with about deadlines 

-  On the Tasks tab, I’d like a little box I can click on and off. I’m thinking the best location is to the right of the time numbers below. This is because it is related to those numbers and I want it super easy to scroll the whole tasks list and see who needs a followup or not.
- It should be visible and usable by every class but Volunteers. Any class checking the box will change its appearance for all others that can see it. This is so everyone knows if a person has been followed up with yet.
- It should have mouseover text that says, “This box is used to track follow ups. Clicking it means you’ve checked in with a person that they are on track to meet their deadline.”
- I’d like the box to be Red outline with light pink filler when a person is at 50% of their task deadline, again at 75% of their task deadline, and again when they are at 90% of their task deadline. This is to make it clear that it needs to be checked/the person needs to be followed up with. Once checked, the box should change back to normal but with a check in it. 
- I’d like the box to reset/auto-clear the check within the box when a person reaches 75% and 90%, so the box is ready/needs to be checked again. This is so I can track when I’ve followed up with people and also know when/if my managers are following up with people.
- I’d like an “i” icon for information that explains all of the information above.
- I’d also like mouseover text that will be different for the box at each level 
      **A -**. 0-50%: Check this box once you’ve checked in for the first time with this team member to make sure they are clear on their task.
      **B -**. 0-50%: Check this box once you’ve checked in for the first time with this team member to make sure they are clear on their task.
      **C -**. Your team member’s task should be at least 75% complete! Check this box once you’ve confirmed they are on track to meet their deadline. Request additional time be added to their task if it is needed.
      **D -**. Your team member’s task should be almost complete! Check this box once you’ve confirmed they are on track to meet their deadline. Request additional time be added to their task if it is needed.
image

## Related PRS:

- This frontend PR is related to the [419](https://github.com/OneCommunityGlobal/HGNRest/pull/419) backend PR.

## Main changes explained:

- Update TeamMemberTask file Add handleCheckboxFollowUp and followUpMouseoverText
- Update TeamMemberTasks file Add handleFollowUp.
- Update TeamMemberTasks/task file Add setFollowUp and checkWhoNeedsFollowUp.
- Update TeamMemberTasks/reducer file Add SET_FOLLOWED_UP.

## How to test:

- check into the current branch.
- do `npm run start:local` to run this PR locally.
- Create a task and assign it to a user. 
- Test for the requirements in the description.

To create a Task for a user, you must be logged in as an Admin or Owner and the individual to be given the task must be assigned the Project the task will be/is a part of. Then go to Other Links—>Projects—>WBS—>Choose the WBS—> Assign the task or “Add Task”  

Here’s a video tutorial too: https://www.loom.com/share/8bc672c323b64955aac4bce46c3794aa?sid=b05f5604-87cb-4ca1-80b8-49bd823893ee  

## Test:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/49083865/9866e33f-6de4-4c81-99e2-779ee6fd178b

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/49083865/0971f0cb-3a03-492e-9621-f6f0b08184fb

## Note:
To observe the changes on the checkbox after adding time to a user, do a hard refresh:
Windows/Linux: Ctrl + F5
Mac: Command + Shift + R

### CODE REVIEW

**Review by cvtqx — DISMISSED** [review]
Checked this PR in Admin, manager and owner accounts. All display this functionality which seems to work as intended.

1. Info modal over i:

info modal

2. Task is 50% complete:

50%complete

3. Task is 75% complete:

75%complete

4. Task is 90% complete:
90%complete

The checkbox, once checked, is reset to pink and unchecked when a new breakpoint is achieved. It is not reset until the 50%,  75% and 90% are achieved. 

We would need to fix the alignment in admin view in a separate PR perhaps:
alignment

**Review by mahamakhan — DISMISSED** [review]
Checked from admin and volunteer account. Looks fine. 


Screenshot 2023-06-23 163958
Screenshot 2023-06-23 164026
Screenshot 2023-06-23 162625

**Comment by rachnapandya** [comment]
The feature is working well.
image
image
image
image

**Review by jonathancarpena — CHANGES_REQUESTED** [review]
@abdel-lall I notice this issue after adding intangible time to a task. After switching the intangible time to tangible, the task progress would not update. I'm not sure if this was intended. 

Test Bug:
1. Log **Intangible Time** to task
2. Update the task from **Intangible** to **Tangible**
3. Check if progress was added after changing time to tangible

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/63978806/14873538-b73d-4139-9abd-38ff4261fbc6

**Comment by one-community** [comment]
> @abdel-lall I notice this issue after adding intangible time to a task. After switching the intangible time to tangible, the task progress would not update. I'm not sure if this was intended.
> 
> Test Bug:
> 
> 1. Log **Intangible Time** to task
> 2. Update the task from **Intangible** to **Tangible**
> 3. Check if progress was added after changing time to tangible

Great catch! This is outside the scope of this PR, please post it to the Bugs Doc and comment on it so I can approve it.

**Comment by abdel-lall** [comment]
@jonathancarpena Well spotted! However, this particular issue falls beyond the scope of the pull request.

**Comment by beblicarl** [comment]
This functionality works as intended

test-5

test-6

test-7

test-8

**Review by xaanders — DISMISSED** [review]
I've tested the PR. Works as intended 

- icon explains the functionality 
i

- hover on the checkbox shows track follow up text, text depends on the task progress

- if progress less 50% checkbox remains unchecked(green). 
before50

- after achieving 50%, 75% and 90% it requires follow up(though checkbox gets unchecked if it was checked before the breakpoint) 
breakpoints

- if followed up then checkbox is checked until next breakpoint
checked


- stays checked when reach 100% and gets unchecked after reaching 75%
100and75

- volunteers see the progress without checkbox
volunteer

**Comment by igorrochadasilva** [comment]
This functionality works as intended. = ) 

- icon explains the functionality
info-modal-over

- Initial checkbox
initial-checkbox

- Volunteer can't see the checkbox
volunteer-cant-see-checkbox


- The checkbox clicked 0%
checkbox-checked-0%

- The checkbox after 50% progress
after-50%-progress

- The checkbox after 50%, when clicked turned green
after-50%-clicked-became-green

- The checkbox after 75% progress
after-75%-progress

- The checkbox after 75%, when clicked turned green
after-75%-clicked-became-green

- The checkbox after 90% progress
after-90%-progress

- The checkbox after 90%, when clicked turned green
after-90%-clicked-became-green

**Review by scoopstroop92 — COMMENTED** [review]
The functionality works as expected . 
image

**Comment by Jiangwei-shi** [comment]
icon explains the functionality
image
After finishing 50%
image
checked
image
After finishing 75%
image
checked
image
After finishing 90%
image
checked
image

**Review by DavidC0126 — COMMENTED** [review]
Hi @abdel-lall, I tested the PR and did not encounter any issue, good work!

-  When the cursor is on  the "i" icon, it will show the explaination.
theIicon

-  In the Initial state, the checked box is green and no check on it.
InitialState


- The functionality works fine in 50%, more than 75%, and over 90% progresses.

progresses_admin

- Volunteer users cannot see the checked box, but are able to see the progresses.

volunteerUser

**Comment by iTvX** [comment]
Checked both admin and owner accounts, and everything works as described. However, I'm experiencing significant lag, and sometimes nothing appears. In such cases, I have to use my mouse to hover over another icon, and then it will reset.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46791134/c3cf0cdd-1ea0-4d84-8250-62b7403ac593

**Review by zuhang2000 — COMMENTED** [review]
Hi I test this PR by following your instructions on checking out both front-end and back-end branch. The overall functionality is working great. One thing is that I'm experiencing extreme high lag when doing this test(it's just that it took a really long time to load the dashboard and all those pop up message). Another thing is that the time can become negative if I edit and delete some of my time log of that task.

Here are my test details. Firstly, the "i" is showing correctly when I move mouse onto it.
Screen Shot 2023-07-29 at 5 02 07 PM

Then, I test all the cases, including when the task progress is below 50%, above 50% but lower than 75%, above 75% but lower than 90%, and at 95%. I check the box whenever it passes the required threshold. It turns out that it can work as expected. When reaching 75% and 90%, the checked box got cleared automatically. 

When I move my mouse onto the checkbox, the message also changed accordingly. The color is also changing according to the progress bar. 



Screen Shot 2023-07-29 at 5 28 40 PM
Screen Shot 2023-07-29 at 5 30 07 PM
In these screenshots, you may see that the box gets automatically cleared when it reaches the threshold.
Screen Shot 2023-07-29 at 5 31 42 PM
Screen Shot 2023-07-29 at 5 33 04 PM

However, I do notice that it's getting strange behaviors if I edit and delete my timelog. It can become negative number. In this case, I must submit more hours of time log in order to compensate this negative time. As you can see in the following screenshots. In order to do the test, I even have to submit a lot more hours. I can also see a lot of negative time records in the "Tasks" list. 
Screen Shot 2023-07-29 at 5 24 13 PM
Screen Shot 2023-07-29 at 5 33 49 PM

**Review by ThapeloMasasa — COMMENTED** [review]
works well as can be seen below.. I do not know why the X symbol is not there though
A)
Screenshot 2023-08-03 at 3 56 50 PM

B)

Screenshot 2023-08-03 at 4 04 57 PM

C)

Screenshot 2023-08-03 at 4 05 06 PM

D)

Screenshot 2023-08-03 at 4 06 20 PM

**Review by winghojackyli — APPROVED** [review]
Hi @abdel-lall , I've tested the PR and it works great. The 50%, 75% and 90% checkpoints have been tested. Besides, I tried using volunteer and it cannot see the checkbox as intended. 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108389005/7354567b-9967-4d4f-b1b9-fc79f1793bd4

**Review by veronicacheng2 — APPROVED** [review]
Hi @abdel-lall, I have tested the PR. The functionality is good but the tasks container is a bit too small and some of the content cannot be shown (the 'i' icon). Other than that, the tracking function works perfectly.

### Green checkbox initially
image

### Red checkbox when reaching over 50%
image

### Checking the box turns it green
image

### Red checkbox when reaching over 75%
image

### Checking the box turns it green
image

**Review by yunshan111 — DISMISSED** [review]
Tested this PR, LTGM.

**Comment by one-community** [comment]
Looking good everyone, keep those reviews and feedback coming. This one is super important!

**Review by Shiwani99 — CHANGES_REQUESTED** [review]
Hi,
I tried to test your PR but when I start the frontend and backend it crashes after login. I'm attaching the screenshot of what is happening in my end. 

Screenshot 2023-09-21 at 6 18 10 PM

**Comment by Shubhu0** [comment]
Tested this PR. Feature working as intended. Can see the icons and all Info icon displays the information as well! And checking the box works but is slow and takes about 2-4 seconds each time to register, but does work and checking the box makes it turn green.

image

image

image

**Review by StrawberryCalpico — CHANGES_REQUESTED** [review]
Hi I tested your PR, however the page will freeze and crash.
Screenshot 2023-10-11 at 6 02 38 PM

**Review by malikjahanzaib — CHANGES_REQUESTED** [review]
Hi @abdel-lall, I tried running this PR locally but it seems to be unresponsive. I am getting a lot of console and BE server errors as well. Refer to my screenshots. Thank you.

Screenshots:

Screenshot 2023-10-27 234037


Screenshot 2023-10-27 233441

Screenshot 2023-10-27 233503

**Review by lichanghaogit — COMMENTED** [review]
Hi @abdel-lall 
all the functions seems work fine for me on local environment using admin and volunteer.
Screenshot 2023-11-17 at 1 35 44 PM
and the infor icon works fine
Screenshot 2023-11-17 at 1 49 29 PM
One thing I am not sure is that when creating new user it will throw me error and stop me from creating new users, is this an issue or should I use the dev site to create user?
Screenshot 2023-11-17 at 1 03 59 PM

**Review by Shuhua-L — CHANGES_REQUESTED** [review]
Update: I ran into the same issue on the dev branch. So, it may not be related to this PR but to the API. I'll come back to this when it's resolved.

---

Hi @abdel-lall, I was able to create a task and assign it to myself. However, when I logged in as a user who had been assigned a task, consistent errors popped up when loading the task, both on the Dashboard page and the TimeLog page. Could you provide some guidance here?

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/83729360/f339bda9-50de-4ffa-b49c-ca27dec3b71a

**Comment by ptrpengdev** [comment]
Hi @abdel-lall, I was able to test some of the functionalities but failed to test with the volunteer role since the Task API failed. 

Env Setup:
Frontend Branch: A[bdel_deadline_follow_up_tracking_solution](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/935)
Backend Branch:  [Abdel_deadline_follow_up_tracking_solution](https://github.com/OneCommunityGlobal/hgnrest/pull/419)

Issue:
Failed to load tasks with the volunteer role.

Step To Reproduce:
Log in with the **volunteer** account: 
  - testtesterpp@gmail.com; password: Hgn@dev2023

Go to Dashboard -> Tasks tab 
   - Team Member Tasks list never finish rendering
   - **Failed API endpoint** with **error** QueryExceededMemoryLimitNoDiskUseAllowed: http://localhost:4500/api/user/6556211f46f10d58d8d9b567/teams/tasks )

Screenshot:
failedToLoadTasksVolunteer

<hr/>
For the Admin role, I was able to test the functionalities:

Icon tooltip **(Sucess)**
Check and uncheck the box **(Sucess)**

Screenshot:
buttonCheckedAdmin
iIconWorksAdmin

**Comment by palakgosalia** [comment]
Hey, I tried testing your PR but the tasks tab wouldn't load. Attached is the screenshot of the error. Hope it helps. Thank you

Env Setup:
Frontend Branch: Ahttps://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/935
Backend Branch: https://github.com/OneCommunityGlobal/hgnrest/pull/419

image

**Comment by kavya151** [comment]
Hi @abdel-lall, I tried to run this PR on my computer locally, and whenever I open the website, the page becomes unresponsive. I've made several attempts, but I consistently encounter the same issue with the webpage freezing.  I didn't face similar problems with other PRs
Screenshot 2023-11-18 at 3 40 25 PM

**Review by sucheta90 — DISMISSED** [review]
@abdel-lall Tested in local as Admin, looks like the app works as intended. Well done.
PR#935_PR#419

**Review by Alforoan — CHANGES_REQUESTED** [review]
All the functionalities seem to work on my end except for volunteer not being able to manipulate the checkbox. As you can see, as a volunteer, I was able to click on the checkbox. Also the checkmark doesn't seem to be right in the center.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/118868270/1041023e-1ef2-4a47-aff5-90cf3982e7c1

**Comment by abdel-lall** [comment]
@Alforoan Hey, I've fixed the issues. When you have a moment, could you please test it again

**Review by Alforoan — CHANGES_REQUESTED** [review]
I keep getting infinite loading error for my tasks and am unable to see any of my tasks.

image

**Comment by one-community** [comment]
> I keep getting infinite loading error for my tasks and am unable to see any of my tasks.

Thank you for your review. This is not related to this PR, it's something else. See the coding problems channel.

**Comment by Haoxiang310** [comment]
I've tested your PR with owner role. I created a new project and assigned a new user to this project. The page of assigning user to the project looks wierd as after I searched and added the new user, the search result disappear after a few second and I cannot delete the user that I just assigned. Also, after entering the WBS page, I cannot see the user I just assigned, thus cannot create a new task. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/135764868/9a3564a2-e2a1-4e81-8c25-9c3142291157

By the way, could you please include the process of creating a task into the PR? Thank you!

**Comment by one-community** [comment]
> I've tested your PR with owner role. I created a new project and assigned a new user to this project. The page of assigning user to the project looks wierd as after I searched and added the new user, the search result disappear after a few second and I cannot delete the user that I just assigned. Also, after entering the WBS page, I cannot see the user I just assigned, thus cannot create a new task.
> 
>  PR7-.935.mov 
> By the way, could you please include the process of creating a task into the PR? Thank you!

Excellent find, please log this as a new bug on the bugs doc and comment on it for my review.

**Comment by one-community** [comment]
I also added how to input a task to the description here

**Review by pshereen — CHANGES_REQUESTED** [review]
After switching to the branch, "Abdel_deadline_follow_up_tracking_solution", tasks are not loading both in the admin dashboard and volunteer dashboard.
image

But the tasks are loading in the dashboard if I switch to development branch.
image

**Review by pshereen — COMMENTED** [review]
Tasks are not loading in the dashboard.

**Comment by yuhuimin1996** [comment]
Hi, I've tested this PR with my owner role, and I met the same problems as @Haoxiang310. After I created a new project and add a volunteer user into it, the user suddenly disappear. And when I click the WBS and try to add assign task, I can't find any buttons or anything like you show in your instructions to create a task. 
Please see more details in the video:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108281570/bbec9a48-06c2-4e69-9fbb-c8d6a118646a

**Review by Sophie-lei — CHANGES_REQUESTED** [review]
I just tested this FE and #419 BE. I met the problem. After I logged in owner role, I could not find other users in the Team Member Tasks. Thus, I can not assign a task to a user. and the console has shown the error as below. It might related to problems in BE #419. 
Screenshot 2024-01-20 at 15 03 20

**Comment by galgalinidhi** [comment]
The Tasks don't load for me either. I cleared the browser data and logged in/out a couple of times but the problem persists. 

Screenshot 2024-01-20 at 10 22 51 PM

**Review by X1aoZhang99 — CHANGES_REQUESTED** [review]
Hi Abdel,

I'm also experiencing difficulties with loading the tasks.

Screenshot 2024-01-22 at 15 37 30

File: src/components/TeamMemberTasks/style.css [hunk 1]
```diff
@@ -53,17 +53,69 @@
...
+  appearance: none;
+  -webkit-appearance: none;
+  -moz-appearance: none;
+  border: 2px solid green;
+  border-radius: 3px;
+  width: 18px;
+  height: 18px;
+  outline: none;
+  background-color: #e1f3e2;
+  /* margin-right: 5px; */
```
Comment (Renanluizssx) [inline src/components/TeamMemberTasks/style.css:91]: you could delete this comment

**Review by Renanluizssx — COMMENTED** [review]
Hi @abdel-lall, i tested your application i noticed some errors, when i checked in some box, it didn't checked, but some of boxes are checked and the message was showed to me very well.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/91035340/5df6408e-bda4-4269-af0a-fb76f3cc9eb5

**Review by SRamen1999 — COMMENTED** [review]
After testing this PR with the BE #419 I'm experiencing difficulties with loading the tasks. I also tested this PR with the current development BE and the task seem to load properly. I think something might be wrong with the BE #419.
Occasionally I would encounter the website page freeze or crash.

Screen Shot 2024-02-08 at 5 29 20 PM

**Review by SaiDeepak-NEU — COMMENTED** [review]
Hello Abdel,

I have tested the changes like mentioned in the PR but din't get the intented results, I don't know if I am doing something wrong or not but here are my findings.
1. I logged in as an admin and navigated to projects and then to wbs of a project and selected a wbs and added a task and assigned a volunteer to that task.
2.  Then I logged in as that volunteer but  in the tasks tab of the dashboard I am not able to see any team member names. The part is not loading at all. I see a blank white space. I tried different things, adding a new task, or editing an already existing task  with more than 3 members assigned to it, but nothing worked. Take a look at the ss of my page
Screenshot (76)
Please take a look or suggest if I am doing anything wrong in here!
Thank you

**Review by XiaohanMeng — COMMENTED** [review]
Hi Abdel, I tested with my Owner and Volunteer account, the same problem appears as @yuhuimin1996  mentioned. There's no button after I create a new task and try to assign it. Also, in the Owner role, the leaderboard and team member tasks don't show member names, which prevents me from following the guidance by clicking the checkbox. Please see them in the video.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113215317/ad6f63eb-aa3b-4925-97f9-117c7865f68c

**Review by brandta-1 — COMMENTED** [review]
# Review
Successfully created a new task for a user in a project as an Admin. When returning to that user's dashboard, the tasks could not load. The error is pictured below, and was a result of the API route ```http://localhost:4500/api/user/65d0f8a1cf23bb6228fa9d32/teams/tasks``` which was called in ```src/actions/task.js```. See [backend review](https://github.com/OneCommunityGlobal/HGNRest/pull/419#pullrequestreview-1887059088) for details on server side.

# Video

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/116298512/ee16d8d0-ea59-41d1-8421-250b89312d10

01_task_in_mongoDB

02_dashboard_task_error

**Comment by nnamdixobi** [comment]
Everything seems to be working okay except for the fact that I was not able to add a user to an existing task like specified in your instructions on how to test

**Review by skaush21 — COMMENTED** [review]
Could not check task feature do to Team members Task not loading.
* logged in using owner account
* waited for page to load, Team members tasks did not load
* console has errors 
* took a look at the code, was unable to determine what was causing the issue. 
task members not loading

console errors

**Review by ClemarNunes — COMMENTED** [review]
The dashboard does not return tasks.

foi

**Review by MALAV1308 — CHANGES_REQUESTED** [review]
Hello , I have tested your PR as per your description. I have created task and assign to the team member but it does not appear in admin account. Something is wrong there from the backend. Need to address this before merging.

**Review by hetvi11110 — CHANGES_REQUESTED** [review]
I've reviewed the PR as mentioned, but I'm unable to check the checkbox even though it's green. Upon clicking the checkbox, it shows me an error message: "Error: Unable to set follow-up." Here is the reference video.



https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/13557067/f860da1f-7b3a-4354-9619-79fadd84589b



Here is what I found in Consol:
image

**Comment by abdel-lall** [comment]
@hetvi11110 the 404 error suggests that the route to api/followup does not exist. It's probably because the backend was not built prior to testing. please make sure to run `npm run build` in the backend after switching to the branch

**Review by hetvi11110 — DISMISSED** [review]
@abdel-lall  I have reviewed it again and it is working fine. Good work!

I just discovered that after unchecking the box, it turned green instead of red on the admin dashboard. Here is the screenshot.
image

**Comment by abdel-lall** [comment]
@hetvi11110 The checkbox doesn't turn red when it's unchecked. Instead, it turns red when users need a follow-up.

Screenshot 2024-04-03 212145

**Review by Shiwani99 — DISMISSED** [review]
Hi,
I reviewed your PR again. it works as expected. I tested for task at 50%, 75%, 90% deadline and the box changed to a different color when these times were recorded. When I checked on the box, it turned to green color. Tooltip also works fine. Great work!

before ticking on the box:
Screenshot 2024-04-04 at 1 27 05 AM

after ticking on the box:
Screenshot 2024-04-04 at 1 27 57 AM

**Review by kylepham — CHANGES_REQUESTED** [review]
PR works as intended. 50%, 75%, and 90% progress breakpoints did trigger the checkbox to be unchecked and displayed in red color. 
Screenshot 2024-04-04 at 2 48 26 PM
Screenshot 2024-04-04 at 2 51 21 PM
Screenshot 2024-04-04 at 2 53 45 PM
Screenshot 2024-04-04 at 2 56 13 PM

**Comment by abdel-lall** [comment]
@kylepham Thank you for the feedback! I've incorporated your suggestion please check it out when you get a chance.

**Review by kylepham — DISMISSED** [review]
Changes look good to me. Nice work 😄 

Screenshot 2024-04-07 at 3 05 13 PM

**Comment by ptrpengdev** [comment]
Hi @abdel-lall

I tested the latest code in this PR. All cases passed except the case - uncheck task with progress > 75% and checkbox should turn red.

PR_935_Review3

In the video, I said 50% but I meant > 75%.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/c554b0e3-6bed-4c19-9ad1-580069fa9a80

**Comment by MQChen211** [comment]
Hi, I tested your PR with the volunteer, admin, and mentor accounts, and it works as intended.

截屏2024-04-10 15 40 21
截屏2024-04-10 15 39 09
截屏2024-04-10 15 35 48

**Comment by abdel-lall** [comment]
Hi @ptrpengdev When a task is assigned to multiple users, the checkbox should turn red for all of them once any of them adds enough tangible time to hit one of the thresholds (50%,75%,90%). That means we need to follow up with all the users working on the task.
I tested it again, and it's working fine for me. Just make sure you hit one of the thresholds when adding time. Thanks!
at 75% and 90%
Screenshot 2024-04-11 173312

Screenshot 2024-04-11 173627

Please test it again when you get a chance. Thank you.

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**